clients/drcachesim/tracer/tracer.cpp

```diff
@@ -404,6 +433,8 @@ instrumentation_init()
         tracing_mode.store(BBDUP_MODE_NOP, std::memory_order_release);
     else if (op_trace_after_instrs.get_value() != 0)
         tracing_mode.store(BBDUP_MODE_COUNT, std::memory_order_release);
+    else if (op_L0_filter_until_instrs.get_value())
+        tracing_mode.store(BBDUP_MODE_L0_FILTER, std::memory_order_release);
 
 #ifdef DELAYED_CHECK_INLINED
     drx_init();
```

Comment:
See event_post_detach below (Github doesn't allow me to add a comment there).

It will probably overwrite tracing_mode, and would cause the issue you're seeing (full tracing if -trace_after_instrs not set).

Reply:
Sorry, I misspoke. I meant s/event_post_detach/event_post_attach/.

Reply:
Sorry but I am still not following which comment you are referring to.

Reply:
> See event_post_detach below (Github doesn't allow me to add a comment there).

Since Github doesn't allow adding comments far from the diff, I couldn't add a comment on the exact event_post_attach line. The current comment thread was supposed to be a proxy to allow discussion.

Anyway, I see you made the fix I was suggesting in  a recent commit (d066b25ba601d747e60d9e446f0977b4c52f37bd). Did it help?

Reply:
Sorry I got confused -- I thought you were referring to some code comment in event_post_attach that described a scenario where tracing_mode would be overwritten. I think you are saying that event_post_attach would be called and it would overwrite tracing_mode. 

But I don't think this function will be called since I am running the application under drrun (`build/bin64/drrun -t drcachesim -L0_filter_until_instrs 10K -trace_for_instrs 10K -outdir traces --offline -verbose 1 -- build/suite/tests/bin/simple_app`) and event_post_attach is only called if we are using source instrumentation (e..g dr_app_start) - is this correct? In any case, the function returns if align_attach_detach_endpoints is false.

Reply:
You're right. This shouldn't be what's  causing the issue you're seeing.